render/renderutils.h

```diff
@@ -16,6 +16,36 @@ namespace pong
     #define GLCall(x) x
 #endif
 
+struct RGBAColor
+{
+    unsigned char r {0};
+    unsigned char g {0};
+    unsigned char b {0};
+    unsigned char a {0};
+};
+
+// Separate RGBAColor struct to avoid converting to opengl's 0.0f - 1.0f range too often
+struct GLRGBAColor
+{
+    float r { 1.0f };
+    float g { 1.0f };
+    float b { 1.0f };
+    float a { 1.0f };
+
+    GLRGBAColor() = default;
+
+    constexpr GLRGBAColor(float r, float g, float b, float a)
```

Comment:
Is there any reason you wouldn't mark these struct constructors with constexpr? In this case I only did because I was using it. constexpr is still a bit unknown to me, I don't see it too often.

Reply:
I think this is the perfect class to do some constexpring with.

Basically what you need to know is constexpr is "this COULD BE evaluated at compile time, if the inputs are known at compile time. In that case, you incur a small compile time cost at the benefit of runtime performance.

This trade off depends on the context, but I think simple dataclasses like this are a general gimme for doing it